src/guide/debugging-models/logging.rst

```diff
@@ -0,0 +1,245 @@
+.. _DebuggingModelsLogging:
+
+Logging Timeseries Data
+=======================
+
+:ref:`Logging<Configuring Data to be Logged>` can be used to collect timeseries data, such as how the size of agent populations change throughout a model's execution. Furthermore, with :ref:`ensembles<ensembles>` this data can be collected across a batch of runs, especially useful when dealing with stochastic models.
+
+Use of logging in this manner is particularly useful when porting a pre-existing model to FLAME GPU 2, and is an approach we have used widely to validate our FLAME GPU 2 models behaviour matches.
+
+1. Decide a model configuration which can be executed in both versions of the model
+2. Execute both models ~100+ times, with a variety of different random seeds.
+3. From these executions, collect timeseries data for a range of important model values. These might be population sizes, environment properties or the mean of agent variables.
+4. Calculate the mean and standard deviation of each data point for both data sets and graph them.
+5. The graph should contain two lines, if the models are operating the same they should match closely (this will also depend on the model's stochasticity and number of runs collected).
+6. Presence of differences in some graphs before others can help narrow down the source of differences assisting in them being solved.
+
+
+Below are some example before and after graphs, demonstrating how visible small bugs can be.
+
+Both the Python and FLAME GPU models were executed with 512 different random seeds, for a low number of steps (60-80).
+
+The number of agents is a major signal for the two models being equivalent.
+
+**The average +- the standard deviation of the number of living NB cells at each time step.**
+
+.. image:: NB_living_count(x512)_a.png
+  :width: 400
+  :alt: The graph shows the initial (bugged) case, whereby the two plots diverge early.
+  
+.. image:: NB_living_count(x512)_b.png
+  :width: 400
+  :alt: The graph shows the final (fixed) case, whereby the two plots match closely.
+  
+When the number of agents is divergent, there will normally be multiple other agent variables which influence agent birth/death which diverge too.
+
+**The average +- the standard deviation of the average of NB agent variable MYCN at each time step.**
```

Comment:
Use an RST heading if it's a heading.

Reply:
Marking this resolved, it's covered by the earlier point.